gravitee-node-kubernetes/src/main/java/io/gravitee/node/kubernetes/keystoreloader/KubernetesPemRegistryKeyStoreLoader.java

```diff
@@ -0,0 +1,197 @@
+                            } else {
+                                String newLocation = String.format(
+                                    "/%s/configmaps/%s",
+                                    currentNamespace,
+                                    items.get(0).getMetadata().getName()
+                                );
+                                this.resources.put(newLocation, ResourceQuery.<ConfigMap>from(newLocation).build());
+
+                                return this.loadKeyStore(items.get(0));
+                            }
+                        })
+                        .doOnComplete(() -> resources.remove(CERTIFICATE_FORMAT_PEM_REGISTRY));
+                }
+
+                return Completable.error(new RuntimeException(String.format("unsupported keystore locations %s", location)));
+            })
+            .toList();
+
+        return Completable.merge(locationObs).andThen(Completable.fromRunnable(this::refreshKeyStoreBundle));
+    }
+
+    @Override
+    protected Flowable<ConfigMap> watch() {
+        final List<Flowable<Event<ConfigMap>>> toWatch = resources
+            .keySet()
+            .stream()
+            .map(location ->
+                kubernetesClient
+                    .watch(WatchQuery.<ConfigMap>from(location).build())
+                    .observeOn(Schedulers.computation())
+                    .repeat()
+                    .retryWhen(errors -> errors.delay(RETRY_DELAY_MILLIS, TimeUnit.MILLISECONDS))
+            )
+            .toList();
+
+        return Flowable.merge(toWatch).filter(event -> event.getType().equalsIgnoreCase("MODIFIED")).map(Event::getObject);
+    }
+
+    @Override
+    protected Completable loadKeyStore(ConfigMap configMap) {
+        if (options.getKeyStoreType().equalsIgnoreCase(CERTIFICATE_FORMAT_PEM_REGISTRY)) {
+            if (
+                configMap.getMetadata().getLabels() != null &&
+                CERTIFICATE_FORMAT_PEM_REGISTRY.equalsIgnoreCase(configMap.getMetadata().getLabels().get(GRAVITEEIO_PEM_REGISTRY_LABEL)) &&
+                configMap.getData() != null
+            ) {
+                return generateKeystoreFromPemRegistry(configMap);
+            } else {
+                // make sure there is no keys left from the past
+                keyStoresByLocation.put(GRAVITEEIO_PEM_REGISTRY_LABEL, initKeyStore());
+                return Completable.complete();
+            }
+        }
+
+        return Completable.error(new RuntimeException(String.format("unsupported keystore type %s", options.getKeyStoreType())));
+    }
+
+    private Completable generateKeystoreFromPemRegistry(ConfigMap configMap) {
+        return Maybe
+            .merge(
+                configMap
+                    .getData()
+                    .values()
+                    .stream()
+                    .flatMap(value -> Stream.of(value.replace("[", "").replace("]", "").replace("\"", "").trim().split(",")))
```

Comment:
You should probably parse that JSON Data, I'm there is an object mapper not too far..